packages/graphql-yoga/src/pubsub.ts

```diff
@@ -77,19 +88,25 @@ export const createPubSub = <
         id === undefined ? routingKey : `${routingKey}:${id as number}`
 
       const { pushValue, asyncIterableIterator } =
-        makePushPullAsyncIterableIterator()
-
-      const handler = (event: Event) => {
-        pushValue((event as any).data)
+        makePushPullAsyncIterableIterator<
+          TPubSubPublishArgsByKey[TKey][1] extends undefined
+            ? TPubSubPublishArgsByKey[TKey][0]
+            : TPubSubPublishArgsByKey[TKey][1]
+        >()
+
+      function pubsubEventListener(
```

Comment:
Added a name for this(functions with names are helpful in debugging - there isn't another reason to do that)